### PR TITLE
Change to ignore  two dictionaries with different key order 

### PR DESCRIPTION
**Pull request by OlteanuRares**
Any serialization/deserialization of json objects can change the key order into a dictionary.
The key order change will count as change and atris will save it.

### CODE REVIEW

File: src/atris/models/helpers.py [hunk 1]
```diff
@@ -15,33 +16,33 @@ def get_diff_fields(model, data, previous_data, excluded_fields_names):
...
+    if field.get_internal_type().strip() == 'JSONField':
+        old = json.loads(old)
+        new = json.loads(new)
+
+    if field.get_internal_type().strip() == 'ArrayField':
+        old = set(old)
+        new = set(new)
+
+    id_list = re.compile(r'^(\d+,\s)+\d+$')
+    new_list = re.match(id_list, new or '')
```
Comment (raresdragomir) [inline src/atris/models/helpers.py:32]: Why are you still using regex matching rather than comparing the objects directly?

Comment (OlteanuRares) [reply src/atris/models/helpers.py:32]: not sure, i don;t know if the intention for this was to cover the only the  arrayfields or other weird text field

Comment (raresdragomir) [reply src/atris/models/helpers.py:32]: I guess @dVeza might be able to confirm this.

Comment (dVeza) [reply src/atris/models/helpers.py:32]: It's meant to detect fk lists serialized at line https://github.com/pbs/django-atris/blob/master/src/atris/models/helpers.py#L65 which before atris 1.4 was unordered.

File: src/atris/models/helpers.py [hunk 2]
```diff
@@ -15,33 +16,33 @@ def get_diff_fields(model, data, previous_data, excluded_fields_names):
...
+    # serializing/deserializing a json object can change key order
+    if field.get_internal_type().strip() == 'JSONField':
+        old = json.loads(old)
+        new = json.loads(new)
+
+    if field.get_internal_type().strip() == 'ArrayField':
+        old = set(old.split(', '))
+        new = set(new.split(', '))
+
+    id_list = re.compile(r'^(\d+,\s)+\d+$')
```
Comment (raresdragomir) [inline src/atris/models/helpers.py:31]: I would simplify the code by dropping the regex matching altogether. Making the "list check" available only for ArrayField type columns. And wait for @dVeza to confirm that this won't break a functionality that they are relying upon.

**Review by andreimutu3 — APPROVED** [review]

**Review by dVeza — COMMENTED** [review]
I suggest keeping the regex, but maybe adding it inside an elif condition like if field.many_to_many or field.one_to_many. just like at serialization. I would also add ArrayField in an elif as opposed to the JsonField just to make the code readable. 
Usually 2 ifs hint that both conditions are possible at the same time.

**Review by raresdragomir — APPROVED** [review]

**Comment by dVeza** [comment]
don't forget to bump the version

**Pull request merged**